esphome/components/api/api_connection.cpp

```diff
@@ -642,8 +642,9 @@ DeviceInfoResponse APIConnection::device_info(const DeviceInfoRequest &msg) {
 }
 void APIConnection::on_home_assistant_state_response(const HomeAssistantStateResponse &msg) {
   for (auto &it : this->parent_->get_state_subs())
-    if (it.entity_id == msg.entity_id)
+    if (it.entity_id == msg.entity_id and it.attribute.value() == msg.attribute) {
```

Comment:
I don't believe we use `and` anywhere else in the codebase. Maybe change for consistency?
```suggestion
    if (it.entity_id == msg.entity_id && it.attribute.value() == msg.attribute) {
```

Reply:
I blame too much Python in my case 😬

Reply:
Haha, yeah I had to double check the file I was reading